From 64661e7451b4ff806670c14bb88c27a21465c780 Mon Sep 17 00:00:00 2001
From: Mao Jinlong <c_jmao@codeaurora.org>
Date: Wed, 12 Jul 2017 15:34:49 +0800
Subject: [PATCH 2/8] DeskClock : Improve the priority of power off alarm
 broadcast

If device power off before power off alarm action is received, power
off alarm value will not be set to rtc. Then this function will not
work. So set power off alarm broadcast to foreground action to make
it be received in time.

CRs-Fixed: 2075052
Change-Id: I6750565396fc79d98e102d2e6515963f3ef978c8
---
 src/com/android/deskclock/alarms/AlarmStateManager.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/com/android/deskclock/alarms/AlarmStateManager.java b/src/com/android/deskclock/alarms/AlarmStateManager.java
index 39f6d0c4f..a49e616af 100644
--- a/src/com/android/deskclock/alarms/AlarmStateManager.java
+++ b/src/com/android/deskclock/alarms/AlarmStateManager.java
@@ -1014,6 +1014,7 @@ public final class AlarmStateManager extends BroadcastReceiver {
     private static void setPowerOffAlarm(Context context, AlarmInstance instance) {
          LogUtils.i("Set next power off alarm : instance id "+ instance.mId);
          Intent intent = new Intent(ACTION_SET_POWEROFF_ALARM);
+         intent.addFlags(Intent.FLAG_RECEIVER_FOREGROUND);
          intent.setPackage(POWER_OFF_ALARM_PACKAGE);
          intent.putExtra(TIME, instance.getAlarmTime().getTimeInMillis());
          context.sendBroadcast(intent);
@@ -1021,6 +1022,7 @@ public final class AlarmStateManager extends BroadcastReceiver {
 
     private static void cancelPowerOffAlarm(Context context, AlarmInstance instance) {
          Intent intent = new Intent(ACTION_CANCEL_POWEROFF_ALARM);
+         intent.addFlags(Intent.FLAG_RECEIVER_FOREGROUND);
          intent.putExtra(TIME, instance.getAlarmTime().getTimeInMillis());
          intent.setPackage(POWER_OFF_ALARM_PACKAGE);
          context.sendBroadcast(intent);
-- 
2.17.1

